File: AlgorithmsProblemSolvingJava/src/io/programminglife/leetcode/Arrays/RemoveDuplicatesFromSortedArray.java
package io.programminglife.leetcode.Arrays;

/**
 * Given a sorted array nums, remove the duplicates in-place such that each element appear only once and return the new length.
 * Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.
 * 
 * Given nums = [1,1,2],
 * Your function should return length = 2, with the first two elements of nums being 1 and 2 respectively.
 * It doesn't matter what you leave beyond the returned length.
 */
public class RemoveDuplicatesFromSortedArray {

    public static int removeDuplicates(int[] nums) {
        if (nums.length == 0) return 0;
        int i = 0;
        for (int j = 1; j < nums.length; j++) {
            if (nums[j] != nums[i]) {
                i++;
                nums[i] = nums[j];
            }
        }
        return i + 1;
    }

    public static void main(String[] args) {
        int[] input1 = {1, 1, 2};
        int[] input2 = {0,0,1,1,1,2,2,3,3,4};
        int[] input3 = {};

        System.out.println(removeDuplicates(input1));
        System.out.println(removeDuplicates(input2));
        System.out.println(removeDuplicates(input3));
    }
    
}

File: AlgorithmsProblemSolvingPython/leetcode/Arrays/removeDuplicatesFromSortedArray.py
def removeDuplicates(nums):
    if len(nums) == 0:
        return 0

    i = 0
    for j in range(len(nums)):
        if nums[j] != nums[i]:
            i = i + 1
            nums[i] = nums[j]

    return i + 1

input1 = [1, 1, 2]
input2 = [0,0,1,1,1,2,2,3,3,4]
input3 = []

print(removeDuplicates(input1))
print("\n")
print(removeDuplicates(input2))
print("\n")
print(removeDuplicates(input3))
print("\n") 